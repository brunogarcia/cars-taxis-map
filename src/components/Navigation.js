import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import BottomNavigation from '@material-ui/core/BottomNavigation';
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
import CarIcon from '@material-ui/icons/DirectionsCar';
import TaxiIcon from '@material-ui/icons/LocalTaxi';
import MapIcon from '@material-ui/icons/Map';

const styles = {
  root: {
    width: 500,
  },
};

class Footer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '',
    };

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event, value) {
    this.setState({
      value,
    });
  }

  render() {
    const { value } = this.state;

    return (
      <BottomNavigation
        value={value}
        onChange={this.handleChange}
        showLabels
      >
        <BottomNavigationAction label="Cars" icon={<CarIcon />} />
        <BottomNavigationAction label="Taxis" icon={<TaxiIcon />} />
        <BottomNavigationAction label="Map" icon={<MapIcon />} />
      </BottomNavigation>
    );
  }
}

export default withStyles(styles)(Footer);
