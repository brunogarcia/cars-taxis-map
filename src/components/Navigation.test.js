import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import Navigation from './Navigation';

test('renders correctly', () => {
  const props = {
    history: {
      push: () => {},
    },
    location: {
      pathname: '/path/to/somewhere',
    },
  };

  const component = renderer.create(<MemoryRouter><Navigation {...props} /></MemoryRouter>);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
