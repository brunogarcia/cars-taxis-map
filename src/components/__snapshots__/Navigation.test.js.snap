// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBottomNavigation-root-2"
>
  <button
    className="MuiButtonBase-root-8 MuiBottomNavigationAction-root-3 MuiBottomNavigationAction-selected-4"
    disabled={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-6"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-11"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-7 MuiBottomNavigationAction-selected-4"
      >
        Cars
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-17"
    />
  </button>
  <button
    className="MuiButtonBase-root-8 MuiBottomNavigationAction-root-3"
    disabled={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-6"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-11"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5H15V3H9v2H6.5c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-7"
      >
        Taxis
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-17"
    />
  </button>
  <button
    className="MuiButtonBase-root-8 MuiBottomNavigationAction-root-3"
    disabled={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiBottomNavigationAction-wrapper-6"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-11"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <g>
          <path
            d="M20.5 3l-.16.03L15 5.1 9 3 3.36 4.9c-.21.07-.36.25-.36.48V20.5c0 .28.22.5.5.5l.16-.03L9 18.9l6 2.1 5.64-1.9c.21-.07.36-.25.36-.48V3.5c0-.28-.22-.5-.5-.5zM15 19l-6-2.11V5l6 2.11V19z"
          />
        </g>
      </svg>
      <span
        className="MuiBottomNavigationAction-label-7"
      >
        Map
      </span>
    </span>
    <span
      className="MuiTouchRipple-root-17"
    />
  </button>
</div>
`;
